packages/lib-core/src/types/extension.ts

```diff
@@ -15,7 +15,6 @@ export type Extension<TType extends string = string, TProperties extends AnyObje
 export type LoadedExtension<TExtension extends Extension = Extension> = TExtension & {
   pluginName: string;
   uid: string;
-  [customProperty: string]: unknown;
```

Comment:
I don't think HAC Dev or HAC Core rely on this logic AFAIK. @vojtechszocs does this have any implications to OCP Console plugins?

Reply:
@florkbr The `Extension` type already has the "allow additional properties whose value is unknown" declaration:

```ts
export type Extension<TType extends string = string, TProperties extends AnyObject = AnyObject> = {
  type: TType;
  properties: TProperties;
  flags?: ExtensionFlags;
  [customProperty: string]: unknown;
};
```

The `LoadedExtension` type is the (augmented) runtime version of the base `Extension` type, it inherits from it via `&` which is why the `[customProperty: string]: unknown` is superfluous in `LoadedExtension`. This is why I removed it.

```ts
export type LoadedExtension<TExtension extends Extension = Extension> = TExtension & {
  pluginName: string;
  uid: string;
};
```

There should be no change at all. You may still add custom properties onto `LoadedExtension` instances.

Reply:
Perfect. I missed that declaration. Thanks!